Observer/RestoreQuote.php

```diff
@@ -168,7 +168,7 @@ public function execute(\Magento\Framework\Event\Observer $observer)
             if ($this->accountConfig->getCartKeepAlive($order->getStore())) {
                 $this->helper->addDebug(__METHOD__ . '|cartKeepAlive enabled|');
 
-                if ($this->checkoutSession->getQuote()
+                if ($this->checkoutSession->getQuote()->getId()
```

Comment:
don't you think it's better to keep $this->checkoutSession->getQuote() for cases when session is zeroized by some reasons and just add extra check $this->checkoutSession->getQuote()->getId() ?